include/RAJA/index/ListSegment.hpp

```diff
@@ -189,27 +281,32 @@ class TypedListSegment
   ///
   template <typename Container>
   explicit TypedListSegment(const Container& container)
-      : m_data(nullptr), m_size(container.size()), m_owned(Unowned)
+    : m_resource(camp::resources::Resource{camp::resources::Host()}),
+      m_use_resource(false),
+      m_owned(Unowned), m_data(nullptr), m_size(container.size())
   {
-    if (m_size <= 0) return;
-    allocate_and_copy<Has_GPU>(container);
-    m_owned = Owned;
+    if (m_size > 0) {
+      allocate_and_copy<Has_GPU>(container);
+      m_owned = Owned;
+    }
   }
 
   ///
   /// Copy-constructor for list segment.
   ///
   TypedListSegment(const TypedListSegment& other)
+    : m_resource(other.m_resource), m_use_resource(other.m_use_resource)
```

Comment:
I know this is pedantic, but providing a default value for all members would be useful

```cpp
, m_owned (Unowned), m_data (nullptr), m_size(0)
```

Reply:
Done!